### PR TITLE
Feature de acordo

### LINKED ISSUE #17: Lógica de resultado do acordo
- [x] Rota de API que recebe o ID do usuário e devolve informações de acordos possíveis, com suas explicações.
- [x] Rota de API que recebe o ID do usuário e informações do acordo para salvar o acordo ao devedor.

### LINKED ISSUE #18: Criação da página de acordo
Criação da página que exibe os três acordos, a partir do consumo da API, e possibilidade de escolher um acordo, ou reformular acordos, se comunicando com a API.

### PR DESCRIPTION
**Pull request by JDaniloC**
## :memo: Summary

Implemented the Proposal Page and its components. closes #18 closes #17

## ✍️  Notes

- 'route.ts' file was created as a stub for validation, therefore must be removed (or replaced by an actual implementation)
- ProposalPodium component isn't responsive yet

### CODE REVIEW

File: nextjs_frontend/src/app/proposal/[cpfDevedor]/components/ProposalCard/proposal-card.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  onSelected,
+}: ProposalCardProps) {
+  function formatJuros(juros: number) {
+    return juros !== 0 ? `${juros * 100}%` : "Sem taxa de juros";
+  };
+
+  function totalValue() {
+    return (proposal.valor / proposal.qtdParcelas)
+            .toFixed(2).replace(".", ",")
+  }
```
Comment (Vinicius-resende-cin) [inline nextjs_frontend/src/app/proposal/[cpfDevedor]/components/ProposalCard/proposal-card.tsx:24]: Shouldn't it be called installmentsValue?

File: nextjs_frontend/src/app/proposal/[cpfDevedor]/components/ProposalPodium/proposal-podium.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+interface ProposalPodiumProps {
+  proposals: Acordo[];
+  changeProposal: (proposal: Acordo) => void;
+}
+
+export default function ProposalPodium({
+  proposals,
+  changeProposal
+}: ProposalPodiumProps) {
+  const [selectedCardID, setSelectedCardID] = useState<number>();
```
Comment (Vinicius-resende-cin) [inline nextjs_frontend/src/app/proposal/[cpfDevedor]/components/ProposalPodium/proposal-podium.tsx:17]: Setting a default selected card makes sense to me as when there is no card featured the proposals appear to be out of order (2nd -> 1st -> 3rd)

**Review by Vinicius-resende-cin — COMMENTED** [review]
Just a few suggestions

File: nextjs_frontend/src/app/proposal/[cpfDevedor]/components/ProposalCard/proposal-card.tsx (hunk 1 above)
Comment (fpeduu) [reply nextjs_frontend/src/app/proposal/[cpfDevedor]/components/ProposalCard/proposal-card.tsx:24]: up

File: nextjs_frontend/src/app/proposal/[cpfDevedor]/components/ProposalCard/proposal-card.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  proposal,
+  position,
+  isSelected,
+  onSelected,
+}: ProposalCardProps) {
+  function formatJuros(juros: number) {
+    return juros !== 0 ? `${juros * 100}%` : "Sem taxa de juros";
+  };
+
+  function totalValue() {
```
Comment (fpeduu) [inline nextjs_frontend/src/app/proposal/[cpfDevedor]/components/ProposalCard/proposal-card.tsx:21]: mudar para algo análogo a valor das parcelas

File: nextjs_frontend/src/app/proposal/[cpfDevedor]/components/ProposalPodium/proposal-podium.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    changeProposal(selectedProposal);
+  };
+
+  return (
+    <div className={Styles.cardWrapper}>
+      {proposalsSorted().map(([position, proposal]) => (
+        <ProposalCard
+          key={proposal.id}
+          proposal={proposal}
+          position={position}
```
Comment (fpeduu) [inline nextjs_frontend/src/app/proposal/[cpfDevedor]/components/ProposalPodium/proposal-podium.tsx:47]: recomendo mudar pra um nome que deixe mais claro o que é, como "priority" ou "optionNumber". "position" deixa meio nebuloso

File: nextjs_frontend/src/app/proposal/api/[cpfDevedor]/route.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import { NextRequest, NextResponse } from "next/server";
+import Acordos, { Acordo } from "@/models/Acordos";
+import { faker } from '@faker-js/faker/locale/pt_BR';
+
+function createRandomAcordo(cpfDevedor: string, minimumValue: number) {
```
Comment (fpeduu) [inline nextjs_frontend/src/app/proposal/api/[cpfDevedor]/route.ts:5]: faz sentido passarmos esse minimumValue? pelo que percebo, no protótipo, isso não aconteceria, apesar de ser uma feature interessante

**Review by fpeduu — CHANGES_REQUESTED** [review]

**Comment by fpeduu** [comment]
Realizar alterações relatadas na dev

**Pull request merged**